src/SFA.DAS.AdminService.Web.Tests/Controllers/Roatp/RoatpFinancialControllerTests.cs

```diff
@@ -27,22 +30,40 @@ public class RoatpFinancialControllerTests
         private RoatpFinancialController _controller;
         private readonly Guid _applicationId = Guid.NewGuid();
         private string _emailAddress = "Test@test.com";
-
+        protected Mock<IHttpContextAccessor> MockHttpContextAccessor;
+        private FinancialReviewDetails _financialReviewDetails;
+       
         [SetUp]
         public void Before_each_test()
         {
             _roatpOrganisationApiClient = new Mock<IRoatpOrganisationApiClient>();
             _applicationApplyApiClient = new Mock<IRoatpApplicationApiClient>();
             _qnaApiClient = new Mock<IQnaApiClient>();
 
+           
+
+            _financialReviewDetails = new FinancialReviewDetails();
+            MockHttpContextAccessor = SetupMockedHttpContextAccessor();
+
             _controller = new RoatpFinancialController(_roatpOrganisationApiClient.Object,
                 _applicationApplyApiClient.Object,
                 _qnaApiClient.Object,
-                Mock.Of<IHttpContextAccessor>());
+                MockHttpContextAccessor.Object)
+            {
+                ControllerContext = new ControllerContext {HttpContext = new DefaultHttpContext()}
+            };
+        }
+
+        private static Mock<IHttpContextAccessor> SetupMockedHttpContextAccessor()
+        {
+            
+            var user = MockedUser.Setup();
 
-            _controller.ControllerContext = new ControllerContext();
-            _controller.ControllerContext.HttpContext = new DefaultHttpContext();
+            var mockHttpContextAccessor = new Mock<IHttpContextAccessor>();
+            var context = new DefaultHttpContext { User = user };
 
+            mockHttpContextAccessor.Setup(_ => _.HttpContext).Returns(context);
+            return mockHttpContextAccessor;
         }
```

Comment:
This looks like a code smell. Controllers don't need a HttpContextAccessor as it's part of the ControllerContext

Reply:
On second look it looks like the controller had this property previously 🤢